trin-core/src/types/header.rs

```diff
@@ -100,8 +90,8 @@ impl Header {
             .append(&self.extra_data);
 
         if with_seal && self.mix_hash.is_some() && self.nonce.is_some() {
-            s.append(&self.mix_hash.unwrap())
-                .append(&self.nonce.unwrap());
+            let nonce = Bytes::copy_from_slice(&self.nonce.unwrap().to_be_bytes());
```

Comment:
Referring to my comment from yesterday: this is another place that suggests to me that we should internally represent the nonce as bytes instead of a u64.